tests/integration/tests/test_ingress.py

```diff
@@ -14,23 +15,35 @@
 def test_ingress(session_instance: List[harness.Instance]):
 
     util.stubbornly(retries=5, delay_s=2).on(session_instance).until(
-        lambda p: "cilium-ingress" in p.stdout.decode()
-    ).exec(["k8s", "kubectl", "get", "service", "-n", "kube-system", "-o", "json"])
+        lambda p: "ingress" in p.stdout.decode()
+    ).exec(["k8s", "kubectl", "get", "service", "-A", "-o", "json"])
 
     p = session_instance.exec(
         [
             "k8s",
             "kubectl",
             "get",
             "service",
-            "-n",
-            "kube-system",
-            "cilium-ingress",
-            "-o=jsonpath='{.spec.ports[?(@.name==\"http\")].nodePort}'",
+            "-A",
+            "-o=json",
         ],
         capture_output=True,
     )
-    ingress_http_port = p.stdout.decode().replace("'", "")
+
+    ingress_http_port = None
+    services = json.loads(p.stdout.decode())
+    for svc in services["items"]:
+        if "ingress" in svc["metadata"]["name"]:
+            LOG.info(f"Found service {svc['metadata']['name']}")
+            for port in svc["spec"]["ports"]:
+                if port["port"] == 80:
+                    ingress_http_port = port.get("nodePort")
```

Comment:
Is there a reason for using .get here instead of []? Is a port not guaranteed to have a nodePort? Not that this is logically wrong, but I'm asking for the sake of consistency with the rest of the dict access code you have.